src/back/spv/writer.rs

```diff
@@ -1510,6 +1594,44 @@ impl Writer {
 
                 RawExpression::Value(id)
             }
+            crate::Expression::ImageLoad {
+                image,
+                coordinate,
+                array_index,
+                index,
+            } => {
+                let image_id =
+                    self.write_expression(ir_module, ir_function, image, block, function)?;
+                let coordinate_id = self.write_texture_coordinates(
+                    ir_module,
+                    ir_function,
+                    coordinate,
+                    array_index,
+                    block,
+                    function,
+                )?;
+
+                let id = self.generate_id();
+                let mut instruction =
+                    Instruction::image_read(id, result_type_id, image_id, coordinate_id);
```

Comment:
It looks like `id` and `result_type_id` are transposed here, and that's causing the invalid output.

Reply:
this is very helpful, thank you!